### ISSUE TITLE
ErrorInvalidB58Checksum Windows 10 Edge

### ISSUE THREAD

**Issue by mareksip** [bitpay/bitcore-lib#55]
Hi,

I am using bitcore-lib within a Cordova project, Android and iOS platforms handle following code OK

    var parent = new bitcore.HDPublicKey("xpub6D2EYhsXPRKz8uvmAntu2p8Hnbyeaq8voLouC6kyko6q3Z7UMtvDpZcFQpqqnZaFFVEVYtvrU27yLc6VjXn4Z1jjLT1sTvvoFKGYE6gnrRx");

but on Windows 10 devices Edge 11, this code fails as:

`Invalid Base58 checksum for xpub6D2EYhsXPRKz8uvmAntu2p8Hnbyeaq8voLouC6kyko6q3Z7UMtvDpZcFQpqqnZaFFVEVYtvrU27yLc6VjXn4Z1jjLT1sTvvoFKGYE6gnrRx`

**Comment by fanatid** [comment]
@mareksip can you run next code in Edge 11?
```js

var bc = require('bitcore-lib')
var buf = bc.encoding.Base58.decode('xpub6D2EYhsXPRKz8uvmAntu2p8Hnbyeaq8voLouC6kyko6q3Z7UMtvDpZcFQpqqnZaFFVEVYtvrU27yLc6VjXn4Z1jjLT1sTvvoFKGYE6gnrRx')
var chsum = bc.crypto.Hash.sha256sha256(buf.slice(0, -4))
console.log(chsum.slice(0, 4).toString('hex'), buf.slice(-4).toString('hex'))
```
I hope it's not [crypto-browserify](http://github.com/crypto-browserify/crypto-browserify) problem.

**Comment by mareksip** [comment]
@fanatid 

I receive long hash in console, I believe this is correct.

Regards, Marek

**Comment by fanatid** [comment]
There shouldn't be long hash.. can you write result here?

**Comment by mareksip** [comment]
@fanatid  Please see updated response.

**Comment by braydonf** [comment]
Related: https://github.com/bitpay/bitcore-lib/issues/53

**Comment by fanatid** [comment]
Hmmm, `buffer.slice` doesn't work???
```js
bc.crypto.Hash.sha256sha256(buf).toString('hex')
// => 3cd1bc9d97f31c600132b4e1d6ff17ba333b5e9a5558cc80bf5f62164b6bf5dd
```

@mareksip for confirming, please run in `Edge 11`:
```js
var buf = new Buffer('Hello World!')
console.log(buf.slice(0, 5).toString())
```
output is `Hello World!` ?

**Comment by mareksip** [comment]
@fanatid well I get

Object Buffer is not defined in IE11 (Windows 8.1. app) probably some library missing?

Regards,
Marek

**Comment by braydonf** [comment]
Browserified Buffer is available at `bitcore.deps.Buffer`

**Comment by mareksip** [comment]
@fanatid @braydonf 

The output is: "Hello"

**Comment by fanatid** [comment]
@braydonf thank you, fixed my comment above

**Comment by fanatid** [comment]
I don't understand... looks like `slice` works fine, but why console output was
```
3cd1bc9d97f31c600132b4e1d6ff17ba333b5e9a5558cc80bf5f62164b6bf5dd
```
instead
```
3cd1bc9d
```

**Comment by fanatid** [comment]
ping @feross

**Comment by feross** [comment]
Edge 11 is old and unsupported by Microsoft. https://en.wikipedia.org/wiki/Microsoft_Edge#Release_history But that other linked issue (https://github.com/bitpay/bitcore-lib/issues/53) shows failure with the latest Edge too.

If this is indeed a buffer issue, it would really help to have a minimal reproducible test case showing what's behaving wrong, so we can add it to the test suite and fix it.

We currently run all of the node.js core buffer tests, straight from their repo, so the tests are *really* good. But I guess it's possible that there's still an issue.

**Comment by braydonf** [comment]
Okay, I updated `bitcore-build` *(that includes browserify dependencies)* locally and created a new browerified build and **all of the 3027 tests passed** in [MSEdge 25.10586.0.0](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/linux/).

**Comment by feross** [comment]
Oh, that's right. There was a bug where Edge wasn't properly supported by buffer a while ago. If you updated to a newer buffer that could have solved your issue.

**Comment by braydonf** [comment]
Thanks @ferros!

Updated browserify build landed in `0.13.15`
